syntax = "proto3";

package dictionary;

service GlossaryService {
  rpc GetAllTerms (Empty) returns (TermsList);
  rpc AddTerm (Term) returns (MessageResponse);
  rpc GetTerm (TermNameRequest) returns (Term);
  rpc UpdateTerm (Term) returns (MessageResponse);
  rpc DeleteTerm (TermNameRequest) returns (MessageResponse);
}

message Empty {}

message Term {
  int32 id = 1;
  string term = 2;
  string definition = 3;
}

message TermsList {
  repeated Term terms = 1;
}

message TermNameRequest {
  string term = 1;
}

message MessageResponse {
  string message = 1;
}